terra/compute/container.py

```diff
@@ -0,0 +1,140 @@
+import os
+import posixpath
+import ntpath
+from os import environ as env
+import re
+import pathlib
+from tempfile import TemporaryDirectory
+import json
+
+from vsi.tools.python import nested_patch
+
+from terra import settings
+from terra.core.settings import TerraJSONEncoder, filename_suffixes
+from terra.compute import compute
+from terra.compute.base import BaseService
+from terra.logger import getLogger
+logger = getLogger(__name__)
+
+
+class ContainerService(BaseService):
```

Comment:
Why did you name this file compute.py instead of singularity.py? It seems like that would line up with docker.py

Reply:
Nevermind I see that this is a parent class of both docker and singularity.

Reply:
It's named container.py, not compute.py. This contained similaritied that docker and singularity both use. Container is not actually a "compute arch", it's a super class.

If you are asking why it the class called `ContainerService` instead of `Service`, it is because this is not a compute, but a super, class, and this differentiates that point.